pkg/controllers/cache/cache.go

```diff
@@ -284,6 +284,58 @@ func (jc *jobCache) TaskCompleted(jobKey, taskName string) bool {
 	return completed >= taskReplicas
 }
 
+func (jc *jobCache) TaskFailed(jobKey, taskName string) bool {
+	jc.Lock()
+	defer jc.Unlock()
+
+	var taskReplicas, retried, maxRetry int32
+
+	jobInfo, found := jc.jobs[jobKey]
+	if !found {
+		return false
+	}
+
+	taskPods, found := jobInfo.Pods[taskName]
+
+	if !found {
+		return false
+	}
+
+	if jobInfo.Job == nil {
+		return false
+	}
+
+	for _, task := range jobInfo.Job.Spec.Tasks {
+		if task.Name == taskName {
+			maxRetry = task.MaxRetry
+			taskReplicas = task.Replicas
+			break
+		}
+	}
+	if taskReplicas <= 0 || maxRetry == -1 {
```

Comment:
Why `taskReplicas ` can be less than 0 and `maxRetry` will be -1?

Reply:
maxRetry == -1 means no retry limit；

Reply:
> maxRetry == -1 means no retry limit；

OK. If that it's better to add some comments